src/utils.rs

```diff
@@ -218,6 +218,45 @@ pub fn collect_prob_dist<E: Event>(
     Ok(prob_dist)
 }
 
+pub fn filter_by_threshold<E: Event>(
```

Comment:
Ok, now we need a good doc comment above. I promise, this is the last request!

Reply:
Don't promise something you might not be able to keep... ;)

I appreciate the code review very much, so please do request further changes if you see something!